### ISSUE TITLE
Integrate Threema Gateway

### ISSUE THREAD

**Issue by rugk** [zammad/zammad#221]
[Threeema Gateway](https://gateway.threema.ch) is a secure messaging API for users of the [Threema app](https://threema.ch/), which is available for many platforms.
It is especially widespread in German-speaking countries.

Messages are end-to-end encrypted and users get it directly on your phone. In contrast to SMS it is way cheaper, you can send longer messages and customize your Gateway channel (own logo, own ID).
Open-source SDKs are available.

**Comment by singular-error** [comment]
i'd love to see that, too. And, if you've got a Threema-ID, you no more need a email-adress. So i would be happy if a new ticket doesn't expect an email, but also accepts a threema-id or a telephone number or anything else unique ;)

**Comment by rugk** [comment]
Yes users should automatically be identified by the thing they provide. This can be a mail, a phone number or a Threema ID, where the latter makes I possible for users to contract the support team without providing personal details as Threema IDs are anonymous.

**Comment by rugk** [comment]
FYI a [rust implementation](https://github.com/dbrgn/threema-gateway-rs) of the Threema Gateway is also being created…

**Comment by thorsteneckel** [comment]
Hi @all - I'm happy to announce that there will be a Threema Gateway integration. I was able to create a Ruby Threema Gateway API client with the friendly and supportive help of the Threema team.

Currently only sending is implemented (which was a critical component for evaluation):
<img src="https://cloud.githubusercontent.com/assets/3873515/23544842/ebe95cba-fff8-11e6-9382-6bb34cff224b.jpg" width="200">

Next step is implementing the receiving functionality which will be done together with the Zammad integration.

Until we have no "muti-identifier" support in Zammad only incoming Threema tickets will be supported.

Please note that this is a side and partly private project so there is no concrete timeline available.

I would be happy to find some beta-testers for this before merging it into stable.

That's it for now 🚀

**Comment by rolfschmidt** [comment]
++

**Comment by rugk** [comment]
Awesome! Will you publish the ruby library (independently) to the integration?

Also I would help beta testing of course.

**Comment by thorsteneckel** [comment]
Hi @rugk - thanks! How can I contact you? You can drop me a mail to support@zammad.com and refer to this issue.

Sure thing! There will be a separate ruby gem and github repository FFA. We ❤️ OpenSource

**Comment by thorsteneckel** [comment]
Just to keep you updated. I published the ruby gem for accessing the Threema Gateway here: https://github.com/thorsteneckel/threema

There are still some minor things todo (see TODO section and code climate reoport) that I will address in the next days.

There is no release yet. I will now start to build the Zammad implementation and keep you updated. 🚀

**Comment by YetAnotherGerrit** [comment]
There are currently no plans to implement that. However, if you still think this would be a good addition to Zammad, please create a feature request in the community.

**Comment by rugk** [comment]
Okay created a post: https://community.zammad.org/t/support-threema-via-threema-gateway/20415?u=rugk
